icu4c/source/i18n/dcfmtsym.cpp

```diff
@@ -329,11 +329,13 @@ struct CurrencySpacingSink : public ResourceSink {
         // both beforeCurrency and afterCurrency were found in CLDR.
         static const char* defaults[] = { "[:letter:]", "[:digit:]", " " };
         if (!hasBeforeCurrency || !hasAfterCurrency) {
-            for (UBool beforeCurrency = 0; beforeCurrency <= true; beforeCurrency++) {
-                for (int32_t pattern = 0; pattern < UNUM_CURRENCY_SPACING_COUNT; pattern++) {
-                    dfs.setPatternForCurrencySpacing((UCurrencySpacing)pattern,
-                        beforeCurrency, UnicodeString(defaults[pattern], -1, US_INV));
-                }
+            for (int32_t pattern = 0; pattern < UNUM_CURRENCY_SPACING_COUNT; pattern++) {
+                dfs.setPatternForCurrencySpacing((UCurrencySpacing)pattern,
+                    false, UnicodeString(defaults[pattern], -1, US_INV));
+            }
+            for (int32_t pattern = 0; pattern < UNUM_CURRENCY_SPACING_COUNT; pattern++) {
+                dfs.setPatternForCurrencySpacing((UCurrencySpacing)pattern,
+                    true, UnicodeString(defaults[pattern], -1, US_INV));
             }
```

Comment:
It looks like this one would have been better as
```
            for (int32_t pattern = 0; pattern < UNUM_CURRENCY_SPACING_COUNT; pattern++) {
                dfs.setPatternForCurrencySpacing((UCurrencySpacing)pattern,
                    false, UnicodeString(defaults[pattern], -1, US_INV));
                dfs.setPatternForCurrencySpacing((UCurrencySpacing)pattern,
                    true, UnicodeString(defaults[pattern], -1, US_INV));
            }
```

Reply:
oh,  I see. I try to minimize any risk of changing the order of the calling sequence since I do not want to assume there are no such dependency. but you are correct that could be shorten with only one loop since dfs.setPatternForCurrencySpacing has no dependency on the seqeunce order of the value passed in.